src/ext/NetFxExtension/wixlib/NetFx47.wxs

```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information. -->
+
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
+
+  <!--
+        .NET Framework installation state properties
+
+        Official documentation can be found at the following location:
+
+        .NET Framework 4.5/4.5.1/4.5.2/4.6/4.6.1/4.6.2/4.7 - http://msdn.microsoft.com/en-us/library/w0x726c2(v=vs.110).aspx
+    -->
+
+  <?define NetFx47MinRelease = 460798 ?>
+  <?define NetFx47WebLink = http://go.microsoft.com/fwlink/?LinkId=825298 ?>
+  <?define NetFx47RedistLink = http://go.microsoft.com/fwlink/?LinkId=825302 ?>
+  <?define NetFx47EulaLink = http://referencesource.microsoft.com/license.html ?>
+
+  <Fragment>
+    <PropertyRef Id="WIXNETFX4RELEASEINSTALLED" />
+    <Property Id="WIX_IS_NETFRAMEWORK_462_OR_LATER_INSTALLED" Secure="yes" />
+    <SetProperty Id="WIX_IS_NETFRAMEWORK_462_OR_LATER_INSTALLED" Value="1" After="AppSearch">
+      WIXNETFX4RELEASEINSTALLED >= "#$(var.NetFx462MinRelease)"
+    </SetProperty>
+  </Fragment>
+
+  <Fragment>
+    <util:RegistrySearchRef Id="NETFRAMEWORK45"/>
+    <util:FileSearch
+      Id='SearchForD3DCompiler'
+      Variable="D3DCompilerExists"
+      Result="exists"
+      Path="%windir%\system32\D3DCompiler_47.dll"
+    />
+
+    <WixVariable Id="WixMbaPrereqPackageId" Value="NetFx47Web" />
+    <WixVariable Id="WixMbaPrereqLicenseUrl" Value="$(var.NetFx47EulaLink)" Overridable="yes" />
+    <WixVariable Id="NetFx47WebDetectCondition" Value="NETFRAMEWORK45 &gt;= $(var.NetFx47MinRelease)" Overridable="yes" />
+    <WixVariable Id="NetFx47WebInstallCondition" Value="" Overridable="yes" />
+    <WixVariable Id="NetFx47WebPackageDirectory" Value="redist\" Overridable="yes" />
+
+    <PackageGroup Id="NetFx47Web">
+      <!-- Update for Windows 7 x64 -->
+      <MsuPackage
+        Id="d3dcompiler64"
+        Compressed="no"
+        Permanent="yes"
+        DetectCondition="((NOT VersionNT64) OR (VersionNT &gt; v6.1) OR (D3DCompilerExists))"
+        DownloadUrl="http://go.microsoft.com/fwlink/?LinkId=848158"
+        KB="KB4019990"
+        Name="redist\Windows6.1-KB4019990-x64.msu">
+        <RemotePayload CertificatePublicKey="371DD003A37769487A2A89A5A9DDB3026451B906" CertificateThumbprint="98ED99A67886D020C564923B7DF25E9AC019DF26" Description="Microsoft .NET Framework 4.7 prerequisite setup" Hash="35CC310E81EF23439BA0EC1F11D7B71DD34ADFE5" ProductName="Microsoft .NET Framework 4.7 prerequisite" Size="2873529" Version="1.0.0.0" />
+      </MsuPackage>
+
+      <!-- Update for Windows 7 x86 -->
+      <MsuPackage
+        Id="d3dcompiler86"
+        Compressed="no"
+        Permanent="yes"
+        DetectCondition="((VersionNT64) OR (VersionNT &gt; v6.1) OR (D3DCompilerExists))"
```

Comment:
DetectCondition should include only D3DCompilerExists
InstallCondition should include the rest of this line.

Also, the package for server isn't included.